### ISSUE TITLE
Scale factor not working on ios

### ISSUE THREAD

**Issue by xuedongyoci** [libretro/RetroArch#16422]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description
Scale factor not work in ozone driver。
So the words look small.

### Expected behavior
Please fix the function，thank you.

### Actual behavior

It doesn't work properly.

### Steps to reproduce the bug

1. Settings-user interface-apperance-scalefactor
2. Change value
3. Nothing happen

### Bisect Results

[Try to bisect and tell us when this started happening]
Version/Commit
You can find this information under Information/System Information

- RetroArch: nightly latest version

### Environment information

- OS:ios iphone 13 pro max
![IMG_1728](https://github.com/libretro/RetroArch/assets/165708176/1dbb7503-1f31-49cd-81bd-d0d76a81b6fd)

- Compiler: [In case you are running local builds]

**Comment by xuedongyoci** [comment]
In the glui driver, It works normally.

**Comment by sonninnos** [comment]
It does work, but it simply has a limit for growing too large width-wise.

**Comment by xuedongyoci** [comment]
> It does work, but it simply has a limit for growing too large width-wise. As in nobody really wants it so big that you can't see and use it.

Please look at the screenshot。
Everything looks too small，I can't even touch it with my fingers.

**Comment by sonninnos** [comment]
Yes, portrait mode width is the limiting factor, and it happens with all platforms, because it is hardcoded in the menu driver. Try smaller values and see that the scaling does in fact work.

**Comment by xuedongyoci** [comment]
> Yes, portrait mode width is the limiting factor, and it happens with all platforms, because it is hardcoded in the menu driver. Try landscape mode and/or smaller values and see that the scaling does in fact work. Sure the limit might be a bit too conservative, but changing it probably causes usability issues in the root page with the sidebar etc.

Thank you, Sonninnos. I understand the limiting factors and it is true that in landscape mode, the font size is normal.
